libvast/builtins/loaders/stdin_loader.cpp

```diff
@@ -0,0 +1,102 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/to.hpp>
+#include <vast/concept/parseable/vast/data.hpp>
+#include <vast/detail/fdinbuf.hpp>
+#include <vast/detail/string.hpp>
+#include <vast/plugin.hpp>
+
+#include <unistd.h>
+
+namespace vast::plugins::stdin_loader {
+
+// -- loader plugin -----------------------------------------------------
+class plugin : public virtual loader_plugin {
+public:
+  static constexpr inline auto max_chunk_size = size_t{16384};
+
+  [[nodiscard]] auto make_loader(const record&, operator_control_plane&) const
+    -> caf::expected<loader> override {
+    return [this]() -> generator<chunk_ptr> {
```

Comment:
The only thing the lambda needs from the plugin instance (as of now) is the `read_timeout` field. It can capture a `read_timeout` copy. -> https://github.com/tenzir/vast/pull/2935/commits/e435401783a132b238d243db154e8e448183d62c